### PR TITLE
buffer video directly into message payload

### PR DESCRIPTION
**Pull request by darenliang**
Also handled case where video is too large to be uploaded.

### CODE REVIEW

File: source/main.py [hunk 1]
```diff
@@ -27,48 +27,71 @@ def error(self, msg):
...
             return
 
-        link = match.group(0)
-        status = match.group(1)
+        for match in matches:
+            cprint(match, blue)
 
-        cprint(link, blue)
+            try:
+                with YoutubeDL(ydl_opts) as ydl:
```
Comment (George-lewis) [inline source/main.py]: Is it a good idea to create a YDL client for each message? I feel like that may have some overhead.

**Comment by darenliang** [comment]
👍

**Comment by George-lewis** [comment]
Please resolve merge conflicts

File: source/main.py (hunk 1 above)
Comment (George-lewis) [reply source/main.py]: I see that you're storing a YDL object inside the bot now -- thank you

**Comment by George-lewis** [comment]
Thanks for your contribution

**Pull request merged**